sdcm/remote/libssh2_client/__init__.py

```diff
@@ -440,25 +440,30 @@ def _process_output(  # pylint: disable=too-many-arguments, too-many-branches
             if perf_counter() > end_time:
                 reader.stop()
                 return False
-            if stdout_stream is not None:
+
+            data_processed = False
+            if stdout_stream is not None and reader.stdout.qsize():
                 try:
                     stdout = reader.stdout.get(timeout=timeout_read_data_chunk)
```

Comment:
@soyacz , does ithis PR realy do what it says ?
I would expect that you would want to make `stdout.get` and `stderr,get` non-blocking, which this PR did not do.

Reply:
I believe `not realiable!` statement is for the size - it's not precise, just approximate - which we don't care, info that there's something is enough.
This PR indeed fixes the issue.

Reply:
Good to hear that, but I still think it would be better to do `.get(timeout=-1)` instead checking on `.qsize()`.

Reply:
under the hood, `.get` also uses similar ways like `qsize` to get the content.